include/revng/Pipeline/Target.h

```diff
@@ -161,7 +119,7 @@ class Target {
   void dumpPathComponents(OStream &OS) const debug_function {
```

Comment:
The `dump` method is not dumping the kind.

Let's have a proper `fromString` and `toString` routines for targets.
Targets are an important actor in the pipeline, let's give them the right importance.

Also, we should either push to the users (e.g., CLI) the parser that understands `*` or, at least, give them a proper name to make sure people does not think that's the standard way to deserialize a target.